osu.Game/Skinning/LegacySkinExtensions.cs

```diff
@@ -100,6 +105,20 @@ IEnumerable<Texture> getTextures(ISkin skin)
             string getFrameName(int frameIndex) => $"{componentName}{animationSeparator}{frameIndex}";
         }
 
+        public static Texture? GetTextureWithMaxSize(this ISkin source, string componentName, Vector2 maxSize, WrapMode wrapModeS = WrapMode.None, WrapMode wrapModeT = WrapMode.None)
```

Comment:
More to the code side of things, why is this a separate method, but `GetAnimation()` gets an optional `maxSize` param? Should be one or the other all the way.

Reply:
I've done some code mashing and removed this overload / new method. See if it reads better to you.

Note that I did this as a prelim requirement. I haven't yet assessed whether the overall direction is correct yet.

Reply:
Just quickly going over source without getting too deep into it, definitely seems much closer to what I'd expect to see.

Reply:
> I've done some code mashing and removed this overload / new method. See if it reads better to you.

It would've been better presenting the change as a proposal first before pushing. I'm not in favour with the new code as it means every skin implementation must respect the `maxSize` parameter & perform the same exact code of trimming the texture, which is better achieved with an extension method rather than providing that as part of the core method itself, especially since the "maximum size" logic can be executed outside of the core method. Not to mention that this massively increases the diff of this PR.

Both of the reasons above are precisely why I haven't considered doing this.

As an alternative, I wouldn't mind something like `skin.GetTexture().WithMaximumSize()` or just keep everything as is and rename `GetTextureWithMaxSize` to `GetTexture`. Whichever feels better (I would prefer the first, because it's very explicit and reads well in code).

Reply:
> it means every skin implementation must respect the `maxSize` parameter & perform the same exact code of trimming the texture

I don't get why this is a concern... If `maxSize` is `null` then the method is a no-op, first of all, so there's no performance consideration there. Secondly, we probably _want_ the max size logic in all implementations for future-proofing. (We were even musing briefly in twitch chat about what happens when user skins are _not_ legacy-texture-based but markup-drawable-based, but dismissed that as unimportant for now.) The diff size concern is like basically irrelevant to me.

Reply:
Well, the change has already been pushed, and if you're both on board with it then I wouldn't bother to mind it.